### PR TITLE
Add support for rebasing

### PR DESCRIPTION
**Pull request by aryzing**
This PR adds support for rebasing through an additional config flag: `integration_tool`. It defaults to `merge`, but also allows for `rebase`. If `rebase` is provided as the `integration_tool`, the PR being worked on will then be rebased onto the base branch before the rest of the CI tasks run.

### CODE REVIEW

**Comment by kveerapen** [comment]
I need this too!

File: git.go [hunk 1]
```diff
@@ -123,6 +124,17 @@ func (g *GitClient) Merge(sha string) error {
 	return nil
 }
 
+// Rebase ...
+func (g *GitClient) Rebase(baseSha string, headSha string) error {
```
Comment (itsdalmo) [inline git.go]: `baseSha` should probably be renamed to `baseRef` as this is the value we are passing as the first argument in `in.go`?

**Comment by aryzing** [comment]
Regarding the format changes, my bad, they were an accidental inclusion, I'll remove them. The name change seems ok with me, will update soon. Thanks for reviewing.

**Comment by aryzing** [comment]
Changes implemented, let me know what you think, thanks for the feedback.

**Pull request merged**